### ISSUE TITLE
How it is possible Theano sum gives these results ?

### ISSUE THREAD

**Issue by fsalmasri** [Theano/Theano#6485]
```
arr = np.ones((2,100,100)).astype(np.float32)x = T.ftensor3('x')
tester = T.sum(x)
tester2k = T.sum(x, axis=2)

s1 = theano.function([x], tester)
s2k = theano.function([x], tester2k)

print s1(arr)
print s2k(arr)

```

The first gives 5625 instead of 20000
The second  gives 30 instead of 100 
[ 30.  30.  30.  30.  30.  30.  30.  30.  30.  30.  30.  30.  30.  30.
   30.  30.  30.  30.  30.  30.  30.  30.  30.  30.  30.  30.  30.  30.
   30.  ..................

**Comment by nouiz** [comment]
For me, I got the good results:
```
>>> print s1(arr)
20000.0

>>> print s2k(arr)
[[ 100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.]
 [ 100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.  100.
   100.  100.  100.  100.]]
>>> 

```

Which version of Theano do you use? Can you update to 0.10.beta4 or the version of github? What are your Theano flag?

**Comment by fsalmasri** [comment]
I'm using Theano 0.9.0b1, Cuda 8 and nvidia driver  375.20  
when testing on CPU i have no problem but when testing on both GPU Titan X and GTX 960 I have these results. 

theano flags:
[global]
device = cuda0 
floatX = float32

[cuda]
root = /usr/local/cuda-8.0
#optimizer_including=cudnn

[nvcc]
flags=-D_FORCE_INLINES
#flags=-arch=sm_52

**Comment by nouiz** [comment]
Update Theano to 0.10beta4. There was changes in some driver that made old
Theano CUDA code generate wrong results. This have been fixed. Your Theano
version is so  old that it probably don't have this fix. I'm not sure if
the fix was done in 0.9, but it was done in the 0.10beta* version.

**Comment by fsalmasri** [comment]
Fixed after updating theano
